docs/compactor.md

```diff
@@ -0,0 +1,93 @@
+1. ETCDBR will take the first full snapshot after it starts main ETCD Process and snapshotter process. But after taking the first full snapshot, snapshotter will only continue taking delta snapshots but no further full snapshots.
+2. Flatten the directory structure of backup folder. Save all the full snapshots, delta snapshots and compacted snapshots under same directory/prefix. Also, name snapshots after revision number instead of timestamps. Restorer will restore from compacted snapshots that have highest revision number in the name and delta snapshots that have higher revision numbers in name than the comapcted full snapshot.
+
+**Proposed Directory Structure**
+```yaml
+Backup :
+    - Full-Snapshot-revision-1
+    - Incremental-Snapshot-revision-6-1 (Format- Incremental-Snapshot-revision-<Revision Number After Last Operation>-<Revision Number Before First Operation>)
+    - Incremental-Snapshot-revision-12-6
+    - Incremental-Snapshot-revision-14-12
+    - Incremental-Snapshot-revision-18-14
+    - Full-Snapshot-revision-18 (Compacted)
+    - Incremental-Snapshot-revision-22-18
+    - Incremental-Snapshot-revision-27-22
+    - Incremental-Snapshot-revision-32-27
+    - Incremental-Snapshot-revision-38-32
+    - Full-Snapshot-revision-38 (Compacted)
+    - Incremental-Snapshot-revision-41-38
+    - Incremental-Snapshot-revision-43-41
+    - Incremental-Snapshot-revision-48-43
+    - Incremental-Snapshot-revision-52-48
+```
+
+3. A meta-data file might need to be added in the Backup directory to mark which compacted (or full) snapshot is ready to be used as the base for restoration. Restorer would study the meta-data file and pickup the ready full snapshot from there and apply all the delta snapshots having revision number greater than that. So, with the backup directory as above, if the content of metadata file is like following:
+
+* Full-Snapshot-revision-1
+* Full-Snapshot-revision-18
+
+Then, during restoration Full-Snapshot-revision-18, and all the delta snaphots from Incremental-Snapshot-revision-22-18 to Incremental-Snapshot-revision-52-48 will be applied. **Restorer would restore from latest ready full snapshot and all subsequent delta snapshots.** As Full-Snapshot-revision-38 is not present in the metadata file, it would not be considered as ready yet.
+
+Another approach to make sure that restorer only picks up snapshots which are ready, full snapshots can be named with extension `.part` until the snapshot is ready. When the snapshot is ready, `.part` extension can be removed from it's name. Restorer would avoid any file having extension as `.part`.  
+
+So, if Full-Snapshot-revision-38 is not ready in the above example, it can be named as `Full-Snapshot-revision-38.part`. Restorer will apply Full-Snapshot-revision-18 and all the delta snaphots from Incremental-Snapshot-revision-22-18 to Incremental-Snapshot-revision-52-48. **Restorer would restore from latest ready full snapshot and all subsequent delta snapshots.**
```

Comment:
Same as above, both, full and delta snapshots need to be **ready**. That's just a nitpick, because you have written `ready` with full snapshot, but not with the delta snapshots.